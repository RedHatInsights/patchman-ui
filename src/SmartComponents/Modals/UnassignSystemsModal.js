import React, { Fragment, useEffect, useState } from 'react';
import propTypes from 'prop-types';
import { Modal, Button, Grid, Skeleton } from '@patternfly/react-core';
import { injectIntl } from 'react-intl';

import messages from '../../Messages';
import { useUnassignSystemsHook } from './useUnassignSystemsHook';
import { renderUnassignModalMessages, filterSystemsWithoutSets } from './Helpers';

const UnassignSystemsModal = ({ unassignSystemsModalState = {}, setUnassignSystemsModalOpen, intl }) => {
    const { systemsIDs, isUnassignSystemsModalOpen } = unassignSystemsModalState;
    const [systemsWithPatchSet, setSystemWithPatchSet] = useState([]);
    const [systemsLoading, setSystemsLoading] = useState(true);

    const handleModalToggle = (shouldRefresh) => {
        setUnassignSystemsModalOpen({
            isUnassignSystemsModalOpen: !isUnassignSystemsModalOpen,
            systemsIDs: [],
            shouldRefresh
        });
    };

    const handleModalClose = () => {
        handleModalToggle(false);
    };

    const handleUnassignment = useUnassignSystemsHook(handleModalToggle, systemsWithPatchSet);

    useEffect(() => {
        setSystemsLoading(true);

        filterSystemsWithoutSets(systemsIDs).then(result => {
            setSystemWithPatchSet(result);
            setSystemsLoading(false);
        });
    }, [systemsIDs]);

    const systemsWithoutPatchSetCount = systemsIDs.length - systemsWithPatchSet.length;

    return (
        <Modal
            variant={'small'}
            isOpen={unassignSystemsModalState.isUnassignSystemsModalOpen}
            title={intl.formatMessage(messages.textUnassignSystemsTitle)}
            onClose={handleModalClose}
            titleIconVariant="warning"
            actions={[
                <Button
                    key="confirm"
                    variant="danger"
                    onClick={handleUnassignment}
                    isDisabled={systemsLoading || systemsWithPatchSet.length === 0}
                >
                    {intl.formatMessage(messages.labelsRemove)}
                </Button>,
                <Button key="cancel" variant="link" onClick={handleModalClose}>
                    {intl.formatMessage(messages.labelsCancel)}
                </Button>
            ]}
        >
            <Grid hasGutter>
                {systemsLoading
                    ? <Skeleton />
                    : <Fragment>
                        {systemsWithPatchSet.length > 0 &&
                            renderUnassignModalMessages('textUnassignSystemsStatement', systemsWithPatchSet.length, intl)
                        }
                        {systemsWithoutPatchSetCount > 0 &&
                            renderUnassignModalMessages('textUnassignSystemsWarning', systemsWithoutPatchSetCount, intl)
                        }
                    </Fragment>
                }
            </Grid>
        </Modal>
    );
};

UnassignSystemsModal.propTypes = {
    intl: propTypes.any,
    setUnassignSystemsModalOpen: propTypes.func,
    unassignSystemsModalState: propTypes.object
};
export default injectIntl(UnassignSystemsModal);
