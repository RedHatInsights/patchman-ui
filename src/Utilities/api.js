import { encodeApiParams } from './Helpers';

export function createApiCall(
    endpoint,
    method,
    parameters = undefined,
    data = undefined
) {
    if (parameters && method === 'get') {
        endpoint = endpoint.concat(encodeApiParams(parameters));
    }

    let result = window.insights.chrome.auth
    .getUser()
    .then(() =>
        fetch('/api/patch/v1' + endpoint, {
            method,
            credentials: 'include',
            body: JSON.stringify(data)
        })
    )
    .then(res => {
        if (!res.ok) {
            const contentType = res.headers.get('content-type');
            if (contentType.indexOf('json') !== -1) {
                throw res;
            } else {
                throw {
                    errors: [
                        { status: res.status, detail: res.statusText }
                    ]
                };
            }
        }

        return res.json();
    })
    .catch(caughtError => {
        const error = Promise.resolve(caughtError || {});
        const genericError = {
            title:
                    'There was an error getting data'
        };
        return error.then(error => {
            const [info] = error.errors;
            const result = error.errors && { ...genericError, detail: info.detail, status: info.status } || genericError;
            throw result ;
        });
    });
    return result;
}

export const fetchApplicableAdvisoriesApi = params => {
    return createApiCall('/advisories', 'get', params);
};

export const fetchApplicableSystemAdvisoriesApi = params => {
    let { id, ...allParams } = params;
    return createApiCall(`/systems/${id}/advisories`, 'get', allParams);
};

export const fetchSystems = params => {
    return createApiCall('/systems', 'get', params);
};

export const fetchAdvisoryDetailsApi = params => {
    return createApiCall(`/advisories/${params.advisoryName}`, 'get');
};

export const fetchApplicablePackagesApi = params => {
    let { id, ...allParams } = params;
    return createApiCall(`/systems/${id}/packages`, 'get', allParams);
};

export const fetchAffectedSystems = params => {
    const { id, ...args } = params;
    return createApiCall(`/advisories/${id}/systems`, 'get', args);
};

export const fetchPackagesList = params => {
    return createApiCall('/packages', 'get', params);
};

export const exportAdvisoriesCSV = params => {
    let endpoint = '/export/advisories';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'text/csv' })
    }).then(res => res.text());
};

export const exportAdvisoriesJSON = params => {
    let endpoint = '/export/advisories';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'application/json' })
    }).then(res => res.json());
};

export const exportSystemsCSV = params => {
    let endpoint = '/export/systems';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'text/csv' })
    }).then(res => res.text());
};

export const exportSystemsJSON = params => {
    let endpoint = '/export/systems';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'application/json' })
    }).then(res => res.json());
};
