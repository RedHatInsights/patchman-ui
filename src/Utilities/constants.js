import BugIcon from '@patternfly/react-icons/dist/js/icons/bug-icon';
import EnhancementIcon from '@patternfly/react-icons/dist/js/icons/enhancement-icon';
import SecurityIcon from '@patternfly/react-icons/dist/js/icons/security-icon';
import React from 'react';
import { subtractDate } from './Helpers';

export const STATUS_REJECTED = 'rejected';
export const STATUS_LOADING = 'loading';
export const STATUS_RESOLVED = 'resolved';

export const compoundSortValues = {
    operating_system: {
        asc: 'os_name,os_major,os_minor',
        desc: '-os_name,-os_major,-os_minor'
    },
    applicable_advisories: {
        asc: 'rhsa_count,rhba_count,rhea_count',
        desc: '-rhsa_count,-rhba_count,-rhea_count'
    }
};

// messy check because of frontend-components tests
export const ENABLE_PACKAGES = window?.insights?.chrome?.isBeta && insights?.chrome?.isBeta();
export const storeListDefaults = {
    rows: [],
    metadata: {
        limit: 25,
        offset: 0,
        total_items: 0
    },
    expandedRows: {},
    selectedRows: {},
    queryParams: {},
    error: {}
};

export const publicDateOptions = [
    { apiValue: '', label: 'All', value: 'all' },
    {
        apiValue: `gt:${subtractDate(7)}`,
        label: 'Last 7 days',
        value: 'last7'
    },
    {
        apiValue: `gt:${subtractDate(30)}`,
        label: 'Last 30 days',
        value: 'last30'
    },
    {
        apiValue: `gt:${subtractDate(90)}`,
        label: 'Last 90 days',
        value: 'last90'
    },
    {
        apiValue: `gt:${subtractDate(365)}`,
        label: 'Last year',
        value: 'lastYear'
    },
    {
        apiValue: `lt:${subtractDate(365)}`,
        label: 'More than 1 year ago',
        value: 'moreThanYear'
    }
];

export const advisorySeverities = [
    {
        value: 0,
        label: 'N/A',
        color: 'var(--pf-global--Color--200)'
    },
    {
        value: 1,
        label: 'Low',
        color: 'var(--pf-global--Color--200)',
        text:
            'This rating is given to all other issues that have a security impact. These are the types of vulnerabilities' +
            ' that are believed to require unlikely circumstances to be able to be exploited, or where a successful exploit' +
            ' would give minimal consequences.'
    },
    {
        value: 2,
        label: 'Moderate',
        color: 'var(--pf-global--warning-color--100)',
        text:
            'This rating is given to flaws that may be more difficult to exploit but could still lead to some' +
            ' compromise of the confidentiality, integrity, or availability of resources, under certain circumstances.' +
            ' These are the types of vulnerabilities that could have had a Critical impact or Important impact' +
            ' but are less easily exploited based on a technical evaluation of the flaw, or affect unlikely' +
            ' configurations.'
    },
    {
        value: 3,
        label: 'Important',
        color: '#ec7a08',
        text:
            'This rating is given to flaws that can easily compromise the confidentiality, integrity, or availability' +
            ' of resources. These are the types of vulnerabilities that allow local users to gain privileges, allow' +
            ' unauthenticated remote users to view resources that should otherwise be protected by authentication,' +
            ' allow authenticated remote users to execute arbitrary code, or allow remote users to cause a denial' +
            ' of service.'
    },
    {
        value: 4,
        label: 'Critical',
        color: 'var(--pf-global--danger-color--100)',
        text:
            'This rating is given to flaws that could be easily exploited by a remote unauthenticated attacker' +
            ' and lead to system compromise (arbitrary code execution) without requiring user interaction.' +
            ' These are the types of vulnerabilities that can be exploited by worms. Flaws that require an authenticated' +
            ' remote user, a local user, or an unlikely configuration are not classed as Critical impact.'
    }
];

export const advisoryTypes = [
    {
        value: 3,
        label: 'Security',
        icon: <SecurityIcon />
    },
    {
        value: 2,
        label: 'Bugfix',
        icon: <BugIcon />
    },
    {
        value: 1,
        label: 'Enhancement',
        icon: <EnhancementIcon />
    }
];

export const updatableTypes = [
    {
        value: '',
        label: 'All'
    },
    {
        value: true,
        label: 'Upgradable'
    }
];

export const packagesListUpdatableTypes = [
    {
        value: '',
        label: 'All'
    },
    {
        value: 'gt:0',
        label: 'Upgradable'
    }
];

export const filterCategories = {
    advisory_type: {
        label: 'Advisory type',
        values: advisoryTypes
    },
    public_date: {
        label: 'Public date',
        values: publicDateOptions
    },
    updatable: {
        label: 'Status',
        values: updatableTypes
    },
    systems_updatable: {
        label: 'Status',
        values: packagesListUpdatableTypes
    }
};

export const entityTypes = {
    advisories: 'advisories',
    packages: 'packages'
};

export const ReadOnlyNotification = {
    title: 'title',
    detail: 'message'
};

export const remediationIdentifiers = {
    package: 'patch-package',
    advisory: 'patch-advisory'
};
