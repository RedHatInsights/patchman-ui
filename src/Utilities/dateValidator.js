import { intl } from './IntlProvider';
import messages from '../Messages';

/*
validates date of a type YYYY-MM-DD with starting limit 1990-01-01.
regex is used to comply the Patternfly date format.
*/
const dateValidator = (dateStr) => {
    const regex = /^(\d{4})-(0[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])$/;

    if (dateStr && typeof dateStr === 'string' && dateStr.match(regex) === null) {
        return intl.formatMessage(messages.labelsErrorInvalidDate);
    }

    const date = new Date(dateStr);
    const timestamp = date.getTime();

    //Month had to be set to equal 0 to get the first month of the year.
    const minDate = new Date(1990, 0, 1);

    if (typeof timestamp !== 'number' || Number.isNaN(timestamp)) {
        return intl.formatMessage(messages.labelsErrorInvalidDate);
    }

    if (date < minDate) {
        return intl.formatMessage(messages.labelsErrorDateLimit);
    }

    return;
};

export default dateValidator;
