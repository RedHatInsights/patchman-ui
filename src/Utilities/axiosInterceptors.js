import {
    authInterceptor,
    responseDataInterceptor
} from '@redhat-cloud-services/frontend-components-utilities/files/interceptors';
import axios from 'axios';
import { ReadOnlyNotification } from './constants';
const axiosInstance = axios.create();

export async function  accessCheckInterceptor (config)  {

    const userPermissions = await insights.chrome.getUserPermissions();
    const inventoryPermissionList = ['inventory:*:*', 'inventory:*:read', 'inventory:hosts:read'];

    const hasInventoryAccess = userPermissions.some((access) => inventoryPermissionList.includes(access?.permission));

    if (!hasInventoryAccess) {
        throw new axios.Cancel('Operation canceled. User does not have access to Inventory');
    }

    return config;
}

export function errorInterceptor(err) {

    if (!axios.isCancel(err)) {

        const { response, isAxiosError } = { ...err };

        if (response && isAxiosError) {
            const { status, statusText, data } = response;

            if (!status) {
                throw err;
            } else {
                const genericError = {
                    title:
                            'There was an error getting data'
                };

                const result = { ...genericError, detail: data.error && data.error || statusText, status };

                throw result;
            }
        }

        throw err;
    }
}

export function readOnlyInterceptor(error) {
    if (error.response && error.response.status === 503) {
        const data = ReadOnlyNotification;
        throw data;
    }

    throw error;
}

axiosInstance.interceptors.request.use(accessCheckInterceptor);
axiosInstance.interceptors.request.use(authInterceptor);
axiosInstance.interceptors.response.use(responseDataInterceptor);
axiosInstance.interceptors.response.use(null, readOnlyInterceptor);
axiosInstance.interceptors.response.use(null, errorInterceptor);

export default axiosInstance;
