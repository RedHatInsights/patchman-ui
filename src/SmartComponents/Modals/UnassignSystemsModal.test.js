import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';
import { Modal } from '@patternfly/react-core';
import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';

import UnassignSystemsModal from './UnassignSystemsModal';
import { removePatchSetApi } from '../../Utilities/api';
import { patchSetUnassignSystemsNotifications } from '../../Utilities/constants';

jest.mock('../../Utilities/api', () => ({
    ...jest.requireActual('../../Utilities/api'),
    removePatchSetApi: jest.fn()
}));

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(() => () => {})
}));

jest.mock('@redhat-cloud-services/frontend-components-notifications/redux', () => ({
    addNotification: jest.fn(() => {})
}));

describe('UnassignSystemsModal', () => {
    let unassignSystemsModalState = {
        isUnassignSystemsModalOpen: true,
        systemsIDs: ['test_1', 'test_2', 'test_3']
    };
    const setUnassignSystemsModalOpen = (modalState) => {
        unassignSystemsModalState = modalState;
    };

    const wrapper = mount(<UnassignSystemsModal
        unassignSystemsModalState={unassignSystemsModalState}
        setUnassignSystemsModalOpen={setUnassignSystemsModalOpen}
    />
    );

    it('should match the snapshots', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should remove systems from a patch set and handle success notification', async ()  => {
        unassignSystemsModalState.isUnassignSystemsModalOpen = true;
        removePatchSetApi.mockReturnValueOnce(
            new Promise((resolve) => {
                resolve({ status: 200 });
            }
            )
        );
        await wrapper.find(Modal).props().actions[0].props.onClick();
        expect(addNotification).toHaveBeenCalledWith(
            patchSetUnassignSystemsNotifications(3).success
        );
        expect(unassignSystemsModalState.isUnassignSystemsModalOpen).toBeFalsy();
    });

    it('should close the modal', () => {
        unassignSystemsModalState.isUnassignSystemsModalOpen = true;
        act(() => {
            wrapper.props().setUnassignSystemsModalOpen(false);
        });

        wrapper.update();
        expect(unassignSystemsModalState.isUnassignSystemsModalOpen).toBeFalsy();
    });

    it('should hide the modal when isUnassignSystemsModalOpen=false', () => {
        const wrapper = mount(<UnassignSystemsModal
            systemsIDs={unassignSystemsModalState.systemsIDs}
            isUnassignSystemsModalOpen={false}
            setUnassignSystemsModalOpen={setUnassignSystemsModalOpen}
        />
        );

        expect(wrapper.find(Modal).props().isOpen).toBeFalsy();
    });
});
