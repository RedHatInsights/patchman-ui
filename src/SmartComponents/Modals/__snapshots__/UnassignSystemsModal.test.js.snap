// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnassignSystemsModal should match the snapshots 1`] = `
<injectIntl(UnassignSystemsModal)
  setUnassignSystemsModalOpen={[Function]}
  unassignSystemsModalState={
    Object {
      "isUnassignSystemsModalOpen": true,
      "systemsIDs": Array [
        "test_1",
        "test_2",
        "test_3",
      ],
    }
  }
>
  <UnassignSystemsModal
    intl={
      Object {
        "$t": [Function],
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "defaultRichTextElements": undefined,
        "fallbackOnEmptyString": true,
        "formatDate": [Function],
        "formatDateTimeRange": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatListToParts": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "default.cancel": "Cancel",
          "default.delete": "Delete",
          "default.remove": "Remove",
          "default.save": "Save",
        },
        "onError": [Function],
        "onWarn": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": undefined,
      }
    }
    setUnassignSystemsModalOpen={[Function]}
    unassignSystemsModalState={
      Object {
        "isUnassignSystemsModalOpen": true,
        "systemsIDs": Array [
          "test_1",
          "test_2",
          "test_3",
        ],
      }
    }
  >
    <Modal
      actions={
        Array [
          <Button
            isDisabled={true}
            onClick={[Function]}
            variant="danger"
          >
            Remove
          </Button>,
          <Button
            onClick={[Function]}
            variant="link"
          >
            Cancel
          </Button>,
        ]
      }
      appendTo={[Function]}
      aria-describedby=""
      aria-label=""
      aria-labelledby=""
      className=""
      hasNoBodyWrapper={false}
      isOpen={true}
      onClose={[Function]}
      ouiaSafe={true}
      showClose={true}
      title="Remove systems from patch set"
      titleIconVariant="warning"
      titleLabel=""
      variant="small"
    >
      <Portal
        containerInfo={
          <div>
            <div
              class="pf-c-backdrop"
            >
              <div
                class="pf-l-bullseye"
              >
                <div
                  aria-describedby="pf-modal-part-2"
                  aria-labelledby="pf-modal-part-1"
                  aria-modal="true"
                  class="pf-c-modal-box pf-m-warning pf-m-sm"
                  data-ouia-component-id="OUIA-Generated-Modal-small-1"
                  data-ouia-component-type="PF4/ModalContent"
                  data-ouia-safe="true"
                  id="pf-modal-part-0"
                  role="dialog"
                >
                  <button
                    aria-disabled="false"
                    aria-label="Close"
                    class="pf-c-button pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      />
                    </svg>
                  </button>
                  <header
                    class="pf-c-modal-box__header"
                  >
                    <h1
                      class="pf-c-modal-box__title pf-m-icon"
                      id="pf-modal-part-1"
                    >
                      <span
                        class="pf-c-modal-box__title-icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 576 512"
                          width="1em"
                        >
                          <path
                            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          />
                        </svg>
                      </span>
                      <span
                        class="pf-u-screen-reader"
                      >
                        Warning alert:
                      </span>
                      <span
                        class="pf-c-modal-box__title-text"
                      >
                        Remove systems from patch set
                      </span>
                    </h1>
                  </header>
                  <div
                    class="pf-c-modal-box__body"
                    id="pf-modal-part-2"
                  >
                    <div
                      class="pf-l-grid pf-m-gutter"
                    >
                      <div
                        class="pf-l-grid__item"
                      >
                        Do you want to remove the 
                        <b>
                           1 
                        </b>
                         selected system  from assigned Patch sets?
                      </div>
                      <div
                        class="pf-l-grid__item"
                      >
                        There  are 
                        <b>
                           2 
                        </b>
                          systems  you are trying to remove that are not assigned to any existing Patch Set. This action will not affect them.
                      </div>
                    </div>
                  </div>
                  <footer
                    class="pf-c-modal-box__footer"
                  >
                    <button
                      aria-disabled="false"
                      class="pf-c-button pf-m-danger"
                      data-ouia-component-id="OUIA-Generated-Button-danger-1"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      Remove
                    </button>
                    <button
                      aria-disabled="false"
                      class="pf-c-button pf-m-link"
                      data-ouia-component-id="OUIA-Generated-Button-link-1"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      Cancel
                    </button>
                  </footer>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <ModalContent
          actions={
            Array [
              <Button
                isDisabled={true}
                onClick={[Function]}
                variant="danger"
              >
                Remove
              </Button>,
              <Button
                onClick={[Function]}
                variant="link"
              >
                Cancel
              </Button>,
            ]
          }
          aria-describedby=""
          aria-label=""
          aria-labelledby=""
          boxId="pf-modal-part-0"
          className=""
          descriptorId="pf-modal-part-2"
          hasNoBodyWrapper={false}
          isOpen={true}
          labelId="pf-modal-part-1"
          onClose={[Function]}
          ouiaId="OUIA-Generated-Modal-small-1"
          ouiaSafe={true}
          showClose={true}
          title="Remove systems from patch set"
          titleIconVariant="warning"
          titleLabel=""
          variant="small"
        >
          <Backdrop>
            <div
              className="pf-c-backdrop"
            >
              <FocusTrap
                active={true}
                className="pf-l-bullseye"
                focusTrapOptions={
                  Object {
                    "clickOutsideDeactivates": true,
                    "tabbableOptions": Object {
                      "displayCheck": "none",
                    },
                  }
                }
                paused={false}
                preventScrollOnDeactivate={false}
              >
                <div
                  className="pf-l-bullseye"
                >
                  <ModalBox
                    aria-describedby="pf-modal-part-2"
                    aria-label=""
                    aria-labelledby="pf-modal-part-1"
                    className="pf-m-warning"
                    data-ouia-component-id="OUIA-Generated-Modal-small-1"
                    data-ouia-component-type="PF4/ModalContent"
                    data-ouia-safe={true}
                    id="pf-modal-part-0"
                    style={Object {}}
                    variant="small"
                  >
                    <div
                      aria-describedby="pf-modal-part-2"
                      aria-label={null}
                      aria-labelledby="pf-modal-part-1"
                      aria-modal="true"
                      className="pf-c-modal-box pf-m-warning pf-m-sm"
                      data-ouia-component-id="OUIA-Generated-Modal-small-1"
                      data-ouia-component-type="PF4/ModalContent"
                      data-ouia-safe={true}
                      id="pf-modal-part-0"
                      role="dialog"
                      style={Object {}}
                    >
                      <ModalBoxCloseButton
                        onClose={[Function]}
                        ouiaId="OUIA-Generated-Modal-small-1"
                      >
                        <Button
                          aria-label="Close"
                          className=""
                          onClick={[Function]}
                          ouiaId="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
                          variant="plain"
                        >
                          <ButtonBase
                            aria-label="Close"
                            className=""
                            innerRef={null}
                            onClick={[Function]}
                            ouiaId="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
                            variant="plain"
                          >
                            <button
                              aria-disabled={false}
                              aria-label="Close"
                              className="pf-c-button pf-m-plain"
                              data-ouia-component-id="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
                              data-ouia-component-type="PF4/Button"
                              data-ouia-safe={true}
                              disabled={false}
                              onClick={[Function]}
                              role={null}
                              type="button"
                            >
                              <TimesIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 352 512"
                                  width="1em"
                                >
                                  <path
                                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                  />
                                </svg>
                              </TimesIcon>
                            </button>
                          </ButtonBase>
                        </Button>
                      </ModalBoxCloseButton>
                      <ModalBoxHeader
                        help={null}
                      >
                        <header
                          className="pf-c-modal-box__header"
                        >
                          <ModalBoxTitle
                            id="pf-modal-part-1"
                            title="Remove systems from patch set"
                            titleIconVariant="warning"
                            titleLabel=""
                          >
                            <h1
                              className="pf-c-modal-box__title pf-m-icon"
                              id="pf-modal-part-1"
                            >
                              <span
                                className="pf-c-modal-box__title-icon"
                              >
                                <ExclamationTriangleIcon
                                  color="currentColor"
                                  noVerticalAlign={false}
                                  size="sm"
                                >
                                  <svg
                                    aria-hidden={true}
                                    aria-labelledby={null}
                                    fill="currentColor"
                                    height="1em"
                                    role="img"
                                    style={
                                      Object {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 576 512"
                                    width="1em"
                                  >
                                    <path
                                      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                                    />
                                  </svg>
                                </ExclamationTriangleIcon>
                              </span>
                              <span
                                className="pf-u-screen-reader"
                              >
                                Warning alert:
                              </span>
                              <span
                                className="pf-c-modal-box__title-text"
                              >
                                Remove systems from patch set
                              </span>
                            </h1>
                          </ModalBoxTitle>
                        </header>
                      </ModalBoxHeader>
                      <ModalBoxBody
                        id="pf-modal-part-2"
                      >
                        <div
                          className="pf-c-modal-box__body"
                          id="pf-modal-part-2"
                        >
                          <Grid
                            container={true}
                            hasGutter={true}
                          >
                            <div
                              className="pf-l-grid pf-m-gutter"
                              container={true}
                            >
                              <Skeleton>
                                <div
                                  className="pf-c-skeleton"
                                >
                                  <span
                                    className="pf-u-screen-reader"
                                  />
                                </div>
                              </Skeleton>
                            </div>
                          </Grid>
                        </div>
                      </ModalBoxBody>
                      <ModalBoxFooter>
                        <footer
                          className="pf-c-modal-box__footer"
                        >
                          <Button
                            isDisabled={true}
                            key="confirm"
                            onClick={[Function]}
                            variant="danger"
                          >
                            <ButtonBase
                              innerRef={null}
                              isDisabled={true}
                              onClick={[Function]}
                              variant="danger"
                            >
                              <button
                                aria-disabled={true}
                                aria-label={null}
                                className="pf-c-button pf-m-danger pf-m-disabled"
                                data-ouia-component-id="OUIA-Generated-Button-danger-1"
                                data-ouia-component-type="PF4/Button"
                                data-ouia-safe={true}
                                disabled={true}
                                onClick={[Function]}
                                role={null}
                                tabIndex={null}
                                type="button"
                              >
                                Remove
                              </button>
                            </ButtonBase>
                          </Button>
                          <Button
                            key="cancel"
                            onClick={[Function]}
                            variant="link"
                          >
                            <ButtonBase
                              innerRef={null}
                              onClick={[Function]}
                              variant="link"
                            >
                              <button
                                aria-disabled={false}
                                aria-label={null}
                                className="pf-c-button pf-m-link"
                                data-ouia-component-id="OUIA-Generated-Button-link-1"
                                data-ouia-component-type="PF4/Button"
                                data-ouia-safe={true}
                                disabled={false}
                                onClick={[Function]}
                                role={null}
                                type="button"
                              >
                                Cancel
                              </button>
                            </ButtonBase>
                          </Button>
                        </footer>
                      </ModalBoxFooter>
                    </div>
                  </ModalBox>
                </div>
              </FocusTrap>
            </div>
          </Backdrop>
        </ModalContent>
      </Portal>
    </Modal>
  </UnassignSystemsModal>
</injectIntl(UnassignSystemsModal)>
`;
